tests/http/test-http-client.php

```diff
@@ -289,4 +290,48 @@ public function test_timeout() {
 				&& 9 === $request->get( 'timeout' )
 		);
 	}
+
+	public function test_sequence() {
+		$this->fake_request(
+			Mock_Http_Response::sequence()
+				->push_status( 200 )
+				->push_status( 400 )
+				->push_status( 500 )
+		);
+
+		$this->assertEquals( 200, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
+		$this->assertEquals( 400, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
+		$this->assertEquals( 500, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
+	}
+
+	public function test_sequence_exception_empty() {
+		$this->expectException( RuntimeException::class );
+		$this->expectExceptionMessage( 'No more responses in sequence.' );
+
+		$this->fake_request(
+			Mock_Http_Response::sequence()
+				->push_status( 200 )
+				->push_status( 400 )
+		);
+
+		$this->assertEquals( 200, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
+		$this->assertEquals( 400, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
+		$this->assertEquals( 500, $this->http_factory->get( 'https://example.com/sequence/' )->status() );
```

Comment:
In this test we're testing that an exception is being thrown when there are no more responses left in the sequence. The test below has a `when_empty()` that will return a different response when empty (rather than an exception). 

That said, this code should be put in `Test_Interacts_With_Requests` instead of `Test_Http_Client` because this is just for a sequence of tests mocks and not actual sequences on the HTTP client. I'll move it.